src/Symfony/Component/AssetMapper/ImportMap/ImportMapVersionChecker.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\AssetMapper\ImportMap;
+
+use Composer\Semver\Semver;
+use Symfony\Component\AssetMapper\Exception\RuntimeException;
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Contracts\HttpClient\Exception\HttpExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class ImportMapVersionChecker
+{
+    private const PACKAGE_METADATA_PATTERN = 'https://registry.npmjs.org/%package%/%version%';
+
+    private HttpClientInterface $httpClient;
+
+    public function __construct(
+        private ImportMapConfigReader $importMapConfigReader,
+        private RemotePackageDownloader $packageDownloader,
+        HttpClientInterface $httpClient = null,
+    ) {
+        $this->httpClient = $httpClient ?? HttpClient::create();
+    }
+
+    /**
+     * @return PackageVersionProblem[]
+     */
+    public function checkVersions(): array
+    {
+        $entries = $this->importMapConfigReader->getEntries();
+
+        $packages = [];
+        foreach ($entries as $entry) {
+            if (!$entry->isRemotePackage()) {
+                continue;
+            }
+
+            $dependencies = $this->packageDownloader->getDependencies($entry->importName);
+            if (!$dependencies) {
+                continue;
+            }
+
+            $packageName = $this->extractPackageNameFromImport($entry->importName);
+
+            $url = str_replace(
+                ['%package%', '%version%'],
+                [$packageName, $entry->version],
+                self::PACKAGE_METADATA_PATTERN
+            );
+            $packages[$packageName] = [
+                $this->httpClient->request('GET', $url),
+                $dependencies,
+            ];
+        }
+
+        $errors = [];
+        $problems = [];
+        foreach ($packages as $packageName => [$response, $dependencies]) {
+            if (200 !== $response->getStatusCode()) {
+                $errors[] = [$packageName, $response];
+                continue;
+            }
+
+            $data = json_decode($response->getContent(), true);
+            // dependencies seem to be found in both places
+            $packageDependencies = array_merge(
+                $data['dependencies'] ?? [],
+                $data['peerDependencies'] ?? []
+            );
+
+            foreach ($dependencies as $dependencyName) {
+                $dependencyVersionConstraint = $packageDependencies[$dependencyName] ?? null;
+
+                if (null === $dependencyVersionConstraint) {
+                    // odd that this would happen
```

Comment:
It makes me nervous :joy:

Reply:
Haha, good point - that comment was a bit ominous. It would be an unexpected but not dangerous situation where we see that a vendor file has `import 'foo'`, but then when we check the metadata for that package, it doesn't list `foo` as a dependency at all. I have no idea if this will ever happen under normal circumstances - I can't think of a way (some misconfigured package ?). 

But, to be safe, I've changed this to also issue a warning (like the other warnings) so the user is aware of it.